src/rng.jl

```diff
@@ -117,6 +117,3 @@ Random123.set_counter!(r::TracedRNG, n::Integer) = r.count = n
 Increase the model step counter by `n`
 """
 inc_counter!(r::TracedRNG, n::Integer=1) = r.count += n
-
-# Handle mutating RNG properly when forking the tape in Libtask
-Libtask.tape_copy(rng::TracedRNG) = copy(rng)
```

Comment:
By default,`Libtask.TapedTask` only deep copies `Array` and `Ref`. See [here](https://github.com/TuringLang/Libtask.jl/blob/master/src/tapedtask.jl#L68) for details. So, we might need to add `TracedRNG` to the `TapedTask` arguments to enable customised deep copying.

Reply:
In addition, we can override the deep and shallow copy behaviour of Julia types by overloading the `tape_shallowcopy` and `tape_deepcopy` functions. See [here](https://github.com/TuringLang/Libtask.jl/blob/master/src/tapedfunction.jl#L456).